server/src/main/java/io/crate/expression/operator/any/AnyRangeOperator.java

```diff
@@ -104,7 +106,16 @@ boolean matches(Object probe, Object candidate) {
     }
 
     @Override
-    protected Query refMatchesAnyArrayLiteral(Function any, Reference probe, @NotNull List<?> nonNullValues, Context context) {
+    protected Query refMatchesAnyArrayLiteral(Function any, Reference probe, Literal<?> candidates, Context context) {
+        if (ArrayType.dimensions(candidates.valueType()) > 1) {
+            return null;
+        }
+        var nonNullValues = StreamSupport
+            .stream(((Iterable<?>) candidates.value()).spliterator(), false)
+            .filter(Objects::nonNull).toList();
+        if (nonNullValues.isEmpty()) {
+            return new MatchNoDocsQuery("Cannot match unless there is at least one non-null candidate");
```

Comment:
Can this be a method in `AnyOperator` and re-used in sub-classes?

Reply:
Thank you, returning `MatchNoDocsQuery` could not be moved to a static method. Please see the updated commit.